python/lsst/summit/utils/utils.py

```diff
@@ -821,13 +822,29 @@ def starTrackerFileToExposure(filename, logger=None):
     # stripped out and somehow not set (plus it doesn't give the midpoint
     # of the exposure), so set it manually from the midpoint here
     try:
+        newArgs = {}  # dict to unpack into visitInfo.copyWith - fill it with whatever needs to be replaced
         md = exp.getMetadata()
+
         begin = datetime.datetime.fromisoformat(md["DATE-BEG"])
         end = datetime.datetime.fromisoformat(md["DATE-END"])
         duration = end - begin
         mid = begin + duration / 2
         newTime = dafBase.DateTime(mid.isoformat(), dafBase.DateTime.Timescale.TAI)
-        newVi = exp.visitInfo.copyWith(date=newTime)
+        newArgs["date"] = newTime
+
+        # AIRPRESS is being set as PRESSURE so afw doesn't pick it up
+        # once we're using the butler for data we will just set it to take
+        # PRESSURE in the translator instead of this
+        weather = exp.visitInfo.getWeather()
+        oldPressure = weather.getAirPressure()
+        if not np.isfinite(oldPressure):
+            pressure = md.get("PRESSURE")
+            if pressure is not None:
+                logger.info("Patching the weather info using the PRESSURE header keyword")
+                newWeather = Weather(weather.getAirTemperature(), pressure, weather.getHumidity())
+                newArgs["weather"] = newWeather
+
+        newVi = exp.visitInfo.copyWith(**newArgs)
```

Comment:
Since you are making this more generic, perhaps wrap this line and the one below it in a `if newArgs:` check.